### PR TITLE
Update Chair, SquareTable, and RectangleTable2 to add chair positioning and create CircleTable2 with chairs around border

### PR DESCRIPTION
**Pull request by rogers-ch**
Chair component updates:

Round Chair style for CircleTable2:
image

Example of chair styles for RectangleTable2 and SquareTable:
image
image

CircleTable2 component:
image

RectangleTable2 component:
image

SquareTable component updates:
image

### CODE REVIEW

**Comment by ralph-dev** [comment]
@rogers-ch can you attach the screenshots in the Pr

File: src/Containers/Chair.tsx [hunk 1]
```diff
@@ -10,65 +13,76 @@ export interface IChair {
      * Boolean value for whether someone is seated in the chair
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:10]: No need to say Boolean value, it's implied by typescript type this is repetition

File: src/Containers/Chair.tsx [hunk 2]
```diff
@@ -10,65 +13,76 @@ export interface IChair {
...
+    /**
+     * Name of Person Sitting on Chair
+     */
+    occupiedBy: string;
+    /**
+     * Determines if the chair will be visible or not
+     */
+    isVisible: boolean;
+    /**
+     * Boolean value for whether the chair is round.
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:26]: Remove Boolean value

File: src/Containers/Chair.tsx [hunk 3]
```diff
@@ -10,65 +13,76 @@ export interface IChair {
...
-    const { colors } = useTheme();
+export const Chair: React.FC<IChair> = ({
+    position = 'top',
+    isSeated = false,
+    occupiedBy = '',
+    isVisible = true,
+    isRound = false,
+    ...props
+}) => {
+    if (isRound) {
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:43]: Put this inside the return statement it looks out of place, 
Do
IsRound ?
GetCurvedCircleChair
:
GetStraightChair

**Review by ralph-dev — COMMENTED** [review]
First round of comments ^

File: src/Containers/Chair.tsx [hunk 4]
```diff
@@ -10,65 +13,76 @@ export interface IChair {
...
 };
 
-interface IBaseChair {
-    chairTableBackground: string;
+/**
+ * This function will determine what color the chair will be
+ * @param isSeated {boolean} - indicated is chair is taken/occupied
+ * @return {string} - Hexadecimal of color
+ */
+function getChairColor(isSeated: boolean) {
```
[the remark this answers is not in the document]
Comment (rogers-ch) [reply src/Containers/Chair.tsx:91]: We're pulling colors from useTheme(). When we tried accessing useTheme() in the styled component, the components wouldn't render in Storybook anymore. Storybook sticks in the loading phase. Is there a particular way to access useTheme() in the styled components?

[the remark this answers is not in the document]
Comment (ralph-dev) [reply src/Containers/Chair.tsx:91]: You cannot useTheme in the styled component

[the remark this answers is not in the document]
Comment (ralph-dev) [reply src/Containers/Chair.tsx:91]: Your supposed to do
${(props) => props.theme}

[the remark this answers is not in the document]
Comment (ralph-dev) [reply src/Containers/Chair.tsx:91]: OR
${({theme}) => theme.Colors etc}

File: src/Containers/CircleTable2.tsx [hunk 5]
```diff
@@ -174,7 +208,11 @@ const ChairWrapper = styled.div<IChairWrapper>`
     width: var(--d);
     height: var(--d);
     --az: calc(
-        ${({ counter }) => counter}*1turn / ${({ numOfChairs }) => numOfChairs}
+        ${({ counter, position, numOfChairs }) =>
+                numOfChairs < 3
```
Comment (ralph-dev) [inline src/Containers/CircleTable2.tsx:216]: This 3 is a magic number

File: src/Containers/CircleTable2.tsx [hunk 6]
```diff
@@ -174,7 +208,11 @@ const ChairWrapper = styled.div<IChairWrapper>`
...
+                    ? getPositionValue(position)
+                    : counter * 1}turn /
+            ${({ numOfChairs }) => (numOfChairs < 3 ? 1 : numOfChairs)}
```
Comment (ralph-dev) [inline src/Containers/CircleTable2.tsx:219]: There is quite a bit of logic why not put it all into a function?
Also if you are doing this much logic, chances are your doing something wrong

**Review by ralph-dev — APPROVED** [review]
LGTM, note how the repo no longer uses the Stories Folder as the Component Folder maintains both.

**Comment by ralph-dev** [comment]
Follow this guide to fix the merge issues btw
https://wolfbeacon.atlassian.net/wiki/spaces/CW/pages/1683030053/3.1+Merge+Conflicts

File: src/Containers/Chair.tsx [hunk 7]
```diff
@@ -105,38 +116,38 @@ const getChairColor: getChairColorType = (isSeated) => {
...
-    border-top-right-radius: 3rem;
-    height: 2rem;
-    width: 10rem;
-    margin-bottom: 0.25rem;
+const topChairStyle = css<Pick<IChair, 'relativeSize'>>`
+    border-top-left-radius: ${({ relativeSize }) => relativeSize * 3}rem;
+    border-top-right-radius: ${({ relativeSize }) => relativeSize * 3}rem;
+    height: ${({ relativeSize }) => relativeSize * 3}rem;
+    width: ${({ relativeSize }) => relativeSize * 10}rem;
+    margin-bottom: ${({ relativeSize }) => relativeSize * 0.25}rem;
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:124]: Group all of these under relative size, way too much variable destructing

File: src/Containers/Chair.tsx [hunk 8]
```diff
@@ -105,38 +116,38 @@ const getChairColor: getChairColorType = (isSeated) => {
...
-    border-bottom-left-radius: 3rem;
-    width: 2rem;
-    height: 10rem;
-    margin: 1.25rem;
+const leftChairStyle = css<Pick<IChair, 'relativeSize'>>`
+    border-top-left-radius: ${({ relativeSize }) => relativeSize * 3}rem;
+    border-bottom-left-radius: ${({ relativeSize }) => relativeSize * 3}rem;
+    width: ${({ relativeSize }) => relativeSize * 3}rem;
+    height: ${({ relativeSize }) => relativeSize * 10}rem;
+    margin: ${({ relativeSize }) => relativeSize * 1.25}rem;
```
Comment (ralph-dev) [inline src/Containers/Chair.tsx:134]: Same issue , group together

File: src/Containers/CircleTable.tsx [hunk 9]
```diff
@@ -19,135 +31,241 @@ export interface ICircleTable {
...
+            case 'Reserved':
                 return colors.occupancyStatusColors.Reserved;
 
-            case occupancyStatusTypes.Occupied:
+            case 'Occupied':
                 return colors.occupancyStatusColors.Occupied;
 
             default:
                 return '';
         }
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:71]: Should this live here or in the styled component? If this only affects styling it should be in styled components

File: src/Containers/CircleTable.tsx [hunk 10]
```diff
@@ -19,135 +31,241 @@ export interface ICircleTable {
...
+            </ChairWrapper>
+        ));
+    };
+
+    /**
+     * Generates a unique key based on a string and a random number
+     * @param pre - a string to append to random number
+     * @returns {string} a unique key
+     */
+    const generateKey: generateKeyType = (pre) => {
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:105]: Key for what
generateChairKeys??

File: src/Containers/CircleTable.tsx [hunk 11]
```diff
@@ -19,135 +31,241 @@ export interface ICircleTable {
...
+                    isVisible={item.isVisible}
+                    isRound={item.isRound}
+                />
+            </ChairWrapper>
+        ));
+    };
+
+    /**
+     * Generates a unique key based on a string and a random number
+     * @param pre - a string to append to random number
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:102]: prefix
not pre?

File: src/Containers/CircleTable.tsx [hunk 12]
```diff
@@ -19,135 +31,241 @@ export interface ICircleTable {
...
+
+type getPositionValueType = (position: Position) => number;
+
+type getTurnValueType = (
+    counter: number,
+    numOfChairs: number,
+    position: Position,
+) => string;
+
+const minChairsBeforeTableResize = 4;
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:150]: MIN_CHAIRS_BEFORE_TABLE_RESIZE
MIN_CHAIRS_TABLE_RESIZE_BREAKPOINT 
use SCREMAING_SNAKE_CASE for consts always when static const (Never Changing)

File: src/Containers/SquareTable/SquareTable.tsx [hunk 13]
```diff
@@ -0,0 +1,333 @@
...
+
+            case 'Reserved':
+                return colors.occupancyStatusColors.Reserved;
+
+            case 'Occupied':
+                return colors.occupancyStatusColors.Occupied;
+
+            default:
+                return '';
+        }
```
Comment (ralph-dev) [inline src/Containers/SquareTable/SquareTable.tsx:195]: Living in the wrong place, Styled component Logic

File: src/Containers/SquareTable/SquareTable.tsx [hunk 14]
```diff
@@ -0,0 +1,333 @@
...
+    chairNumOnSide: number;
+    chairNumOnTop: number;
+    relativeSize: number;
+}
+
+const TableBody = styled.div<ITableBody>`
+    height: ${({ chairNumOnSide, relativeSize }) =>
+        chairNumOnSide * 20 * relativeSize}rem;
+    width: ${({ chairNumOnTop, relativeSize }) =>
+        chairNumOnTop * 20 * relativeSize}rem;
```
Comment (ralph-dev) [inline src/Containers/SquareTable/SquareTable.tsx:282]: Group

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Lots of Styling needs to be grouped together so that
1) You can write LESS code
2) Code is more maintainable

File: src/Containers/CircleTable.tsx [hunk 15]
```diff
@@ -181,46 +156,71 @@ const getPositionValue: getPositionValueType = (position) => {
...
+ * and returns the hexadecimal color value as a string
+ *
+ * @param occupancyStatus - the occupancy status for the table
+ * @return {string} - Hexadecimal color value
+ */
+const getOccupancyColor: getOccupancyColorType = (occupancyStatus) => {
+    switch (occupancyStatus) {
+        case 'Vacant':
+            return useTheme().colors.occupancyStatusColors.Vacant;
+
```
Comment (ralph-dev) [inline src/Containers/CircleTable.tsx:176]: why is there random enter lines

File: src/Containers/SquareTable/SquareTable.tsx [hunk 16]
```diff
@@ -265,6 +234,30 @@ export const SquareTable: React.FC<ISquareTable> = ({
...
+ * Determines the correct color for Status and ColorDiv based on occupancyStatus
+ * and returns the hexadecimal color value as a string
+ *
+ * @return {string} - Hexadecimal color value
+ */
+const getOccupancyColor: getOccupancyColorType = (occupancyStatus) => {
+    switch (occupancyStatus) {
+        case 'Vacant':
+            return useTheme().colors.occupancyStatusColors.Vacant;
+
```
Comment (ralph-dev) [inline src/Containers/SquareTable/SquareTable.tsx:249]: random enter lines?

File: src/Containers/SquareTable/_ChairRow.tsx [hunk 17]
```diff
@@ -159,9 +159,10 @@ interface ISideChairRow {
 const SideChairRow = styled.div<ISideChairRow>`
     display: flex;
     flex-wrap: wrap;
-    margin-right: ${({ relativeSize }) => -15 * relativeSize}px;
-    margin-left: ${({ relativeSize }) => -15 * relativeSize}px;
-    height: ${({ relativeSize }) => 20 * relativeSize}rem;
+    ${({ relativeSize }) =>
+        `margin-right: ${-15 * relativeSize}px;
```
Comment (ralph-dev) [inline src/Containers/SquareTable/_ChairRow.tsx:163]: This -15 is a magic number what does it mean?
WHITESPACE_BETWEEN_CHAIR_AND_TABLE?

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Still Magic Numbers in the Code, not all of them just specific ones

**Comment by ralph-dev** [comment]
Also there is merge conflicts ^, Merge Master into this Branch fix conflicts, then wait for approval so you can merge into master

**Review by ralph-dev — APPROVED** [review]
LGTM we will do the fixes in another PR

**Review by ralph-dev — APPROVED** [review]
LGTM, you can merge

**Pull request merged**